llvm/test/Transforms/InstCombine/sub-xor-cmp.ll

```diff
@@ -0,0 +1,19 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 4
+; RUN: opt < %s -passes=instcombine -S | FileCheck %s
+
+define  i64 @absdiff(i64 %0, i64 %1) {
```

Comment:
What is the motivation of this patch? Does this pattern exist in some real-world applications?

Reply:
> Does this pattern exist in some real-world applications?

Yes. We have observed this pattern in our internal workload.

Reply:
Similar to https://www.oreilly.com/library/view/hackers-delight/0201914654/0201914654_ch02lev1sec4.html.